nwb_conversion_tools/datainterfaces/behavior/movie/moviedatainterface.py

```diff
@@ -152,88 +144,113 @@ def run_conversion(
             f"({len(image_series_kwargs_list)}) vs. file_paths ({len(self.source_data['file_paths'])})!"
         )
 
-        for j, file in enumerate(file_paths):
-            timestamps = starting_times[j] + get_movie_timestamps(movie_file=file)
-
-            if len(starting_times) != len(file_paths):
-                starting_times.append(timestamps[-1])
+        def _check_duplicates(image_series_kwargs_list):
+            image_series_kwargs_list_keys = [i["name"] for i in image_series_kwargs_list]
+            if len(set(image_series_kwargs_list_keys)) < len(image_series_kwargs_list_keys):
+                assert external_mode, "For multiple video files under the same ImageSeries name, use exernal_mode=True."
+            keys_set = []
+            image_series_kwargs_list_unique = []
+            file_paths_list = []
+            for n, image_series_kwargs in enumerate(image_series_kwargs_list):
+                if image_series_kwargs["name"] not in keys_set:
+                    keys_set.append(image_series_kwargs["name"])
+                    file_paths_list.append([file_paths[n]])
+                    image_series_kwargs_list_unique.append(dict(image_series_kwargs))
+                else:
+                    idx = keys_set.index(image_series_kwargs["name"])
+                    file_paths_list[idx].append(file_paths[n])
+            return image_series_kwargs_list_unique, file_paths_list
 
-            image_series_kwargs = dict(image_series_kwargs_list[j])
-            if check_regular_timestamps(ts=timestamps):
-                fps = get_movie_fps(movie_file=file)
-                image_series_kwargs.update(starting_time=starting_times[j], rate=fps)
-            else:
-                image_series_kwargs.update(timestamps=H5DataIO(timestamps, compression="gzip"))
+        image_series_kwargs_list_updated, file_paths_list = _check_duplicates(image_series_kwargs_list)
+        if starting_times is not None:
+            assert isinstance(starting_times, list) and all(
+                [isinstance(x, float) for x in starting_times]
+            ), "Argument 'starting_times' must be a list of floats."
+            assert len(starting_times) == len(image_series_kwargs_list_updated), (
+                "starting times list length must be equal number of unique ImageSeries " "containers to write to nwb"
+            )
+        else:
+            starting_times = [0.0] * len(image_series_kwargs_list_updated)
 
+        for j, (image_series_kwargs, file_list) in enumerate(zip(image_series_kwargs_list_updated, file_paths_list)):
             if external_mode:
-                image_series_kwargs.update(format="external", external_file=[file])
+                with VideoCaptureContext(str(file_list[0])) as vc:
+                    fps = vc.get_movie_fps()
+                image_series_kwargs.update(
+                    starting_time=starting_times[j], rate=fps, format="external", external_file=file_list
+                )
```

Comment:
If https://github.com/catalystneuro/nwb-conversion-tools/pull/372/files#diff-57b78875c81d904caa745de93f7771de666b821e9d52586cd03a8328e7a74c6bR173 is triggered here, all the `starting_times` for all the separate `ImageSeries` will be `0.0`.

The logic of the previous interface was to at least assume that the videos were consecutive, and adjusting the `starting_times` of each so that they start when the last one ended. You can calculate the time of that by also pulling the total number of frames and dividing by the `fps`, which I would do at initialization (https://github.com/catalystneuro/nwb-conversion-tools/pull/372/files#diff-57b78875c81d904caa745de93f7771de666b821e9d52586cd03a8328e7a74c6bR173) instead of setting them all to `0.0`.

Reply:
I think its better for us to have starting times as Nans instead of 0s if they are not specified. Its more accurate than assuming that videos are continuous.